cpp/examples/symreg/CMakeLists_standalone.txt

```diff
@@ -0,0 +1,139 @@
+#
+# Copyright (c) 2021, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+cmake_minimum_required(VERSION 3.8 FATAL_ERROR)
+include(ExternalProject)
+
+project(symreg_example VERSION 0.1.0 LANGUAGES CXX CUDA )
+
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+find_package(CUDAToolkit)
+
```

Comment:
You can now do `find_package(cuml)` and get proper cuml targets and dependencies which removes the need for the majority of this file. 

Really roughly this is what you need:

```cmake

find_package(CUDAToolkit)
find_package(cuml)

add_executable(symreg_example symreg_example.cpp)
# Need to set linker language to CUDA to link the CUDA Runtime
set_target_properties(symreg_example PROPERTIES LINKER_LANGUAGE "CUDA")

# Link cuml and cudart
target_link_libraries(symreg_example PRIVATE cuml::cuml++ CUDA::cudart)

```

Reply:
This is really slick!